pegasus/src/social_metadata.rb

```diff
@@ -0,0 +1,160 @@
+# Centralized social metadata for a few key pages:
+#   code.org/
+#   csedweek.org/
+#   hourofcode.com/
+#   code.org/challenge
+
+def get_social_metadata_for_page(request)
+  # Important:
+  #   - image should always come before video
+  #   - description should always come before description_twitter
+  social_tags = {
+    "code.org" => {
+      "soon-hoc" => {
+        title: I18n.t(:og_title),
+        description: I18n.t(:og_description),
+        image: {path: "/images/fit-1220/social-media/default-og-image.jpg", width: 1220, height: 640}
+      },
+      "week-before-hoc" => {
+        title: I18n.t(:og_title_soon),
+        description: I18n.t(:og_description_celeb),
+        image: {path: "/images/fit-1220/social-media/celeb-challenge.jpg", width: 1220, height: 640}
+      },
+      "actual-hoc" => {
+        title: I18n.t(:og_title_here),
+        description: I18n.t(:og_description_celeb),
+        image: {path: "/images/fit-1220/social-media/celeb-challenge.jpg", width: 1220, height: 640}
+      },
+      "default" => {
+        title: I18n.t(:og_title),
+        description: I18n.t(:og_description),
+        image: {path: "/images/fit-1220/social-media/default-og-image.jpg", width: 1220, height: 640},
+        video: {youtube_key: "nKIu9yen5nc", width: 640, height: 360},
+      }
+    },
+    "csedweek.org" => {
+      "soon-hoc" => {
+        title: I18n.t(:csedweek_og_title),
+        description: I18n.t(:csedweek_og_description_soon),
+        image: {path: "/images/cs-is-everything-thumbnail.png", width: 1200, height: 627},
+        video: {youtube_key: "QvyTEx1wyOY", width: 640, height: 360}
```

Comment:
Worth extracting a descriptive constant for `{youtube_key: "QvyTEx1wyOY", width: 640, height: 360}`?

Reply:
Same feedback for `{youtube_key: "KsOIlDT145A", width: 640, height: 360}` below.

Reply:
I thought about that, but tended towards duplication for now to make it a bit easier to reason through the table without having to refer to anything outside of it.  Neutral on that decision though.

Reply:
I think `video: CS_IS_CHANGING_EVERYTHING` may be a little clearer?  Otherwise I have to look up https://www.youtube.com/watch?v=QvyTEx1wyOY to see which video is referenced.

Reply:
Oh that's a great point.  Will do that.